webapp/src/userSettings.ts

```diff
@@ -1,23 +1,95 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {notifySettingsChanged} from './nativeApp'
+import {Utils} from './utils'
+
+// eslint-disable-next-line no-shadow
+enum UserSettingKey {
+    Language = 'language',
+    Theme = 'theme',
+    LastBoardId = 'lastBoardId',
+    LastViewId = 'lastViewId',
+    EmojiMartSkin = 'emoji-mart.skin',
+    EmojiMartLast = 'emoji-mart.last',
+    EmojiMartFrequently = 'emoji-mart.frequently',
+    RandomIcons = 'randomIcons'
+}
+
 export class UserSettings {
+    static get(key: UserSettingKey): string | null {
+        return localStorage.getItem(key)
+    }
+
+    static set(key: UserSettingKey, value: string | null) {
```

Comment:
Love the idea of abstracting UserSettings away from localStorage. One thing to be careful of is localStorage is used for other things too (like the session token). Might be a good idea to check that key is always in the set of allowable keys defined above.

Also, I noticed in `importUserSettings`, that it stamps entries into localStorage. Might be good to check the keys there too. I can't think of any adverse security issues, but this might open holes in the future.

Reply:
Ah, yes that's an excellent point. I added an `if` to check for the allowed keys in both places.